lib/ZcapClient.js

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {DEFAULT_HEADERS, httpClient} from '@digitalbazaar/http-client';
+import {signCapabilityInvocation} from 'http-signature-zcap-invoke';
+
+export class ZcapClient {
+  /**
+   * An object that can make HTTPS requests such as node's https.Agent or ky.
+   *
+   * @typedef {object} httpsAgent
+   * @see https://nodejs.org/api/https.html#https_class_https_agent
+   *
+   * Creates a new ZcapClient instance that can be used to perform
+   * Authorization Capability (ZCAP) requests against HTTP URLs.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} options.baseUrl - The base URL for the client to use
+   *   when building invocation request URLs.
+   * @param {object} options.defaultHeaders - The default HTTP headers to
+   *   include in every invocation request.
+   * @param {object} options.agent - An optional HttpsAgent to use to
+   *   when performing HTTPS requests.
+   * @param {object} options.invocationSigner - An object that contains a
+   *   `.sign()` function and key `id` parameter that will be used for
+   *   signing the request.
+   *
+   * @returns {ZcapClient}.
+   */
+  constructor(
+    {baseUrl, defaultHeaders = {}, agent, invocationSigner} = {}) {
+    this.baseUrl = baseUrl;
+    this.agent = agent;
+    this.defaultHeaders = {...DEFAULT_HEADERS, ...defaultHeaders};
+    this.invocationSigner = invocationSigner;
```

Comment:
Not sure if strongly binding the `invocationSigner` to the client will cover our use cases -- needs more investigation.

Reply:
I'm guessing that we can optionally override the invocationSigner per call, it'll be a parameter override option.